src/frontend/src/handler/create_sink.rs

```diff
@@ -291,6 +298,96 @@ pub fn gen_sink_plan(
     })
 }
 
+// This function is used to return partition compute info for a sink. More details refer in `PartitionComputeInfo`.
+// Return:
+// `Some(PartitionComputeInfo)` if the sink need to compute partition.
+// `None` if the sink does not need to compute partition.
+pub async fn get_partition_compute_info(
+    with_options: &WithOptions,
+) -> Result<Option<PartitionComputeInfo>> {
+    let properties = HashMap::from_iter(with_options.clone().into_inner().into_iter());
+    let Some(connector) = properties.get(UPSTREAM_SOURCE_KEY) else {
+        return Ok(None);
+    };
+    match connector.as_str() {
+        ICEBERG_SINK => {
+            let iceberg_config = IcebergConfig::from_hashmap(properties)?;
+            get_partition_compute_info_for_iceberg(&iceberg_config).await
+        }
+        _ => Ok(None),
+    }
+}
+
+async fn get_partition_compute_info_for_iceberg(
+    iceberg_config: &IcebergConfig,
+) -> Result<Option<PartitionComputeInfo>> {
+    let table = create_table(iceberg_config).await?;
+    let Some(partition_spec) = table.current_table_metadata().current_partition_spec().ok() else {
+        return Ok(None);
+    };
+
+    if partition_spec.is_unpartitioned() {
+        return Ok(None);
+    }
+
+    // Separate the partition spec into two parts: sparse partition and range partition.
+    // Sparse partition means that the data distribution is more sparse at a given time.
+    // Range partition means that the data distribution is likely same at a given time.
+    // Only compute the partition and shuffle by them for the sparse partition.
```

Comment:
so we are changing the distribution key to adjust each CN's hash range?

Reply:
Yes, we can see that for sparse partition, we will add the exchange before sink.
https://github.com/risingwavelabs/risingwave/blob/8b263cb9a4471386f05ddd0239440a4a54ece73b/src/frontend/planner_test/tests/testdata/output/sink.yaml#L233